synapse/storage/monthly_active_users.py

```diff
@@ -182,6 +182,18 @@ def upsert_monthly_active_user(self, user_id):
         Args:
             user_id (str): user to add/update
         """
+        # Support user never to be included in MAU stats. Note I can't easily call this
+        # from upsert_monthly_active_user_txn because then I need a _txn form of
+        # is_support_user which is complicated because I want to cache the result.
+        # Therefore I call it here and ignore the case where
+        # upsert_monthly_active_user_txn is called directly from
+        # _initialise_reserved_users reasoning that it would be very strange to
+        #  include a support user in this context.
```

Comment:
I'd feel happier about this if you actually checked that the user wasn't a support user in `_initialise_reserved_users` and made sure that there was a warning about it on the docstring of `upsert_monthly_active_user_txn`.

Alternatively, you could assume that the overhead of doing the is_support_user check is negligible compared to the MAU upsert, and define an `is_support_user_txn` (without a cache) which you can call from upsert_monthly_active_user_txn.

Reply:
I have added a check in _initialise_reserved_users using a txn version is_support, I've also updated the docstring on upsert_monthly_active_user_txn

I want to avoid this check being an uncached call from upsert_monthly_active_user_txn because upsert_monthly_active_user_txn is called frequently and is_support is almost always going to be False.